backend/layers/processing/utils/dask_utils.py

```diff
@@ -0,0 +1,25 @@
+import logging
+
+import tiledb
+from scipy import sparse
+
+logger = logging.getLogger(__name__)
+
+
+class TileDBSparseArrayWriteWrapper:
+    def __init__(self, uri, *, ctx=None):
+        self.uri = uri
+        self.ctx = {
+            "sm.io_concurrency_level": 1,
+            "sm.compute_concurrency_level": 1,
+        }
```

Comment:
It's possible that these parameters are worth tuning. I had forced these to 1 in my testing so that I could make sure I wouldn't have oversubscription problems.

Reply:
Looking at the tiledb documentation the default setting should be sufficient.